templates/common/config/00-config.conf

```diff
@@ -60,6 +60,9 @@ username={{ .ServiceUser }}
 {{ if (index . "ServicePassword") }}
 password = {{ .ServicePassword }}
 {{ end }}
+{{ if (index . "MemcachedServersWithInet") }}
+memcached_servers = {{ .MemcachedServersWithInet }}
+{{ end }}
```

Comment:
Without the `if (index . "MemcachedServersWithInet")` I get:
```
2024-02-25T17:09:29-05:00	ERROR	Reconciler error	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "GlanceAPI": {"name":"glance-default-single","namespace":"openstack"}, "namespace": "openstack", "name": "glance-default-single", "reconcileID": "60084dbf-994d-4bd6-8471-139180efea4f", "error": "template: tmp:63:23: executing \"tmp\" at <.MemcachedServersWithInet>: map has no entry for key \"MemcachedServersWithInet\""}
```
With this it works for me in that I see the property getting set in my configuration:
```
[fultonj@hamfast glance-operator{memcached}]$ oc get secret glance-config-data -o json | jq -r '.data."00-config.conf"' | base64 -d | grep memcached
memcached_servers = inet:[memcached-0.memcached]:11211
[fultonj@hamfast glance-operator{memcached}]$ 
```

Reply:
interesting, I have to check this one because it shouldn't happen

Reply:
ok I think I understand now. It makes sense to have the if condition over there because we rely on the same template (`00-config.conf`) that can be used both in `glance_controller` (by cronjobs, dbsync and other resources), and in `glanceapi_controller` (which is the one that is failing for you). In particular, each `glanceapi` controller build an extended version of that config file, hence we set multiple templates options that are missing there.
I'm wondering if this option is available in `glance-<apiName>-config-data`, otherwise we need to propagate it to the sublevel CR.
Right now I think we're not using memcached at all because the config option is not propagated to the underlying subCR.

[1] https://github.com/openstack-k8s-operators/glance-operator/blob/main/controllers/glanceapi_controller.go#L843

Reply:
I think this issue is resolved. The `memcached_servers` are now where they should be. Thanks for your help.

```
[fultonj@hamfast functional{memcached}]$ oc get secret glance-config-data -o json | jq -r '.data."00-config.conf"' | base64 -d | grep memcached
[fultonj@hamfast functional{memcached}]$ oc get secret glance-default-single-config-data -o json | jq -r '.data."00-config.conf"' | base64 -d | grep memcache
memcached_servers = inet:[memcached-0.memcached]:11211
[fultonj@hamfast functional{memcached}]$ 
```